sdk/clientcore/core/src/main/java/io/clientcore/core/credentials/oauth/AccessTokenType.java

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package io.clientcore.core.credentials.oauth;
+
+import io.clientcore.core.utils.ExpandableEnum;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Represents Access Token types.
+ */
+public final class AccessTokenType implements ExpandableEnum<String> {
+    private static final Map<String, AccessTokenType> VALUES = new ConcurrentHashMap<>();
+    private final String caseSensitive;
+    private final String caseInsensitive;
+
+    private AccessTokenType(String name) {
+        this.caseSensitive = name;
+        this.caseInsensitive = name.toLowerCase();
+    }
+
+    @Override
+    public String getValue() {
+        return caseSensitive;
+    }
+
+    /**
+     * Gets the {@link AccessTokenType} based on the name passed into {@link #fromString(String)}.
+     *
+     * @return The token type based on the construction of this {@link AccessTokenType}.
+     */
+    public String getCaseSensitiveName() {
+        return toString();
+    }
+
+    /**
+     * Gets the {@link AccessTokenType} lower cased.
+     *
+     * @return The {@link AccessTokenType} lower cased.
+     */
+    public String getCaseInsensitiveName() {
+        return caseInsensitive;
+    }
+
```

Comment:
I think these are an optimization in `HttpHeaderName` and maybe we don't need them here? @alzimmermsft?

Reply:
TokenType can be case sensitive depending on how strict the implementation is on the user/service  end.